core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -74,28 +74,22 @@ private <T extends Identity> SdkHttpFullRequest sraSignRequest(SdkHttpFullReques
 
         ExecutionAttributes executionAttributes = context.executionAttributes();
         SelectedAuthScheme<T> selectedAuthScheme =
-            executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+            (SelectedAuthScheme<T>) executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
 
         if (!shouldSign(selectedAuthScheme)) {
             return request;
         }
 
         Pair<SdkHttpFullRequest, Duration> measuredSign = MetricUtils.measureDuration(
             () -> {
-                AuthSchemeOption authSchemeOption = selectedAuthScheme.authSchemeOption();
-
-                // TODO: Identity resolution should move to before Endpoint resolution interceptor, to support accountId based
-                //  endpoints and also to logically separate out identity resolution as its own step.
-                ResolveIdentityRequest.Builder identityRequestBuilder = ResolveIdentityRequest.builder();
-                authSchemeOption.forEachIdentityProperty(identityRequestBuilder::putProperty);
-
-                IdentityProvider<T> identityProvider = selectedAuthScheme.identityProvider();
-                T identity = identityProvider.resolveIdentity(identityRequestBuilder.build()).join();
+                CompletableFuture<T> identityFuture =
+                    (CompletableFuture<T>) executionAttributes.getAttribute(SdkInternalExecutionAttribute.IDENTITY);
```

Comment:
If you store the identity in the SelectedAuthScheme instead of a separate execution attribute, you can ensure the identity type matches the one in the SelectedAuthScheme. That means you won't have to cast it, because it'll already be T when you retrieve it.